src/blockchain/src/blockchain.rs

```diff
@@ -14,6 +14,8 @@
    limitations under the License.
 */
 
+use identity::ed25519::Keypair;
+use identity::PublicKey::Ed25519;
 use libp2p::identity;
```

Comment:
Can we please not add indirection like this?

```suggestion
use libp2p::identity{self, ed25519::Keypair, PublicKey::Ed25519};
```

Reply:
What do you mean by `indirection`?